### ISSUE TITLE
ALE not detected phpcs errors because lines ended on ^M.

### ISSUE THREAD

**Issue by ta-tikoma** [dense-analysis/ale#3111]
## Information

**VIM version**

nvim 0.4.3

Operating System: Windows 10

## What went wrong

ALE not detected phpcs errors because lines ended on ^M.
I add echom:
![1](https://user-images.githubusercontent.com/2934128/79470659-1f8c9280-801b-11ea-9a9c-c7a68b3d4ed0.jpg)
and see:
![2](https://user-images.githubusercontent.com/2934128/79470823-4ba81380-801b-11ea-8688-32855e2a3b66.jpg)
may be change regex to `^.*:\(\d\+\):\(\d\+\): \(.\+\) - \(.\+\) (\(.\+\))` ?

### :ALEInfo

```
 Current Filetype: php
Available Linters: ['langserver', 'phan', 'php', 'phpcs', 'phpmd', 'phpstan', 'psalm']
  Enabled Linters: ['php', 'phpcs']
 Suggested Fixers: 
  'php_cs_fixer' - Fix PHP files with php-cs-fixer.
  'phpcbf' - Fix PHP files with phpcbf.
  'remove_trailing_lines' - Remove all blank lines at the end of a file.
  'trim_whitespace' - Remove all trailing whitespace characters at the end of every line.
 Linter Variables:

let g:ale_php_php_executable = 'php'
let g:ale_php_phpcbf_standard = 'PSR12'
let g:ale_php_phpcbf_use_global = 1
let g:ale_php_phpcs_executable = 'phpcs'
let g:ale_php_phpcs_options = '-'
let g:ale_php_phpcs_standard = ''
let g:ale_php_phpcs_use_global = 1
 Global Variables:

let g:ale_cache_executable_check_failures = v:null
let g:ale_change_sign_column_color = 0
let g:ale_command_wrapper = ''
let g:ale_completion_delay = v:null
let g:ale_completion_enabled = 0
let g:ale_completion_max_suggestions = v:null
let g:ale_echo_cursor = 1
let g:ale_echo_msg_error_str = 'Error'
let g:ale_echo_msg_format = '%code: %%s'
let g:ale_echo_msg_info_str = 'Info'
let g:ale_echo_msg_warning_str = 'Warning'
let g:ale_enabled = 1
let g:ale_fix_on_save = 1
let g:ale_fixers = {'php': ['phpcbf', 'trim_whitespace', 'remove_trailing_lines']}
let g:ale_history_enabled = 1
let g:ale_history_log_output = 1
let g:ale_keep_list_window_open = 0
let g:ale_lint_delay = 200
let g:ale_lint_on_enter = 1
let g:ale_lint_on_filetype_changed = 1
let g:ale_lint_on_insert_leave = 1
let g:ale_lint_on_save = 1
let g:ale_lint_on_text_changed = 'normal'
let g:ale_linter_aliases = {}
let g:ale_linters = {'php': ['php', 'phpcs']}
let g:ale_linters_explicit = 0
let g:ale_list_vertical = 0
let g:ale_list_window_size = 10
let g:ale_loclist_msg_format = '%code: %%s'
let g:ale_lsp_root = {}
let g:ale_max_buffer_history_size = 20
let g:ale_max_signs = -1
let g:ale_maximum_file_size = v:null
let g:ale_open_list = 0
let g:ale_pattern_options = v:null
let g:ale_pattern_options_enabled = v:null
let g:ale_set_balloons = 0
let g:ale_set_highlights = 1
let g:ale_set_loclist = 1
let g:ale_set_quickfix = 0
let g:ale_set_signs = 1
let g:ale_sign_column_always = 0
let g:ale_sign_error = '>>'
let g:ale_sign_info = '--'
let g:ale_sign_offset = 1000000
let g:ale_sign_style_error = '>>'
let g:ale_sign_style_warning = '--'
let g:ale_sign_warning = '--'
let g:ale_sign_highlight_linenrs = 0
let g:ale_statusline_format = v:null
let g:ale_type_map = {}
let g:ale_use_global_executables = v:null
let g:ale_virtualtext_cursor = 0
let g:ale_warn_about_trailing_blank_lines = 1
let g:ale_warn_about_trailing_whitespace = 1
  Command History:

(executable check - success) php
(finished - exit code 0) 'cmd /s/c "php -l -d error_reporting=E_ALL -d display_errors=1 -d log_errors=0 -- < C:\Users\torad\AppData\Local\Temp\nvimDHDToX\1\Test.php"'

<<<OUTPUT STARTS>>>
No syntax errors detected in Standard input code
<<<OUTPUT ENDS>>>

(executable check - success) phpcs
(finished - exit code 2) 'cmd /s/c "cd /d C:\Work && phpcs -s --report=emacs --stdin-path=C:\Work\Test.php - < C:\Users\torad\AppData\Local\Temp\nvimDHDToX\3\Test.php"'

<<<OUTPUT STARTS>>>
C:\Work\Test.php:1:1: error - End of line character is invalid; expected "\n" but found "\r\n" (Generic.Files.LineEndings.InvalidEOLChar)
C:\Work\Test.php:2:1: error - Missing file doc comment (PEAR.Commenting.FileComment.Missing)
C:\Work\Test.php:5:1: error - Missing doc comment for class Test (PEAR.Commenting.ClassComment.Missing)
<<<OUTPUT ENDS>>>
```
